### PR TITLE
[androiddebugbridge] Add mDNS discovery for android tv

### PR DESCRIPTION
**Pull request by GiviMAD**
Signed-off-by: Miguel Álvarez <miguelwork92@gmail.com>
Add mDNS discovery participant for android tv.

### CODE REVIEW

**Comment by GiviMAD** [comment]
Tested with Chromecast with Android TV and a Xiaomi tv with Android TV.

File: bundles/org.openhab.binding.androiddebugbridge/README.md [hunk 1]
```diff
@@ -20,9 +20,11 @@ Please update this document if you tested it with other android versions to refl
 
 ## Discovery
 
-As I can not find a way to identify android devices in the network the discovery will try to connect through adb to all the reachable ip in the defined range.
+Android TV and Fire TV devices should be discovered automatically (using mDNS).
 
-You could customize the discovery process through the binding options. 
+As I can not find other android a way to identify other android devices in the network the manual discovery scan will try to connect through adb to all the reachable ip in the defined range.
```
Comment (lsiepel) [inline bundles/org.openhab.binding.androiddebugbridge/README.md:25]: Please update this line to somehting like: The binding is unable to....

Comment (GiviMAD) [reply bundles/org.openhab.binding.androiddebugbridge/README.md]: Text updated!

File: bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/discovery/AndroidTVMDNSDiscoveryParticipant.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link AndroidTVMDNSDiscoveryParticipant} is responsible for discovering new and removed Android TV devices. It
+ * uses
+ * the central {@link MDNSDiscoveryService}.
+ *
+ * @author Miguel Álvarez - Initial contribution
+ */
+@Component(configurationPid = "discovery.androiddebugbridge")
```
Comment (lolodomo) [inline bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/discovery/AndroidTVMDNSDiscoveryParticipant.java:47]: Some bindings also set `service = MDNSDiscoveryParticipant.class` but not all.
Unfortunately, I don't know what is the recommendation.
So this is just a comment, not a request for change.

Comment (lolodomo) [reply bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/discovery/AndroidTVMDNSDiscoveryParticipant.java:47]: Maybe @wborn or @jlaur you know what is the recommendation here ?

Comment (GiviMAD) [reply bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/discovery/AndroidTVMDNSDiscoveryParticipant.java]: I have added it to both services as seems appropriate.

**Review by lolodomo — APPROVED** [review]
LGTM, thank you

**Comment by lolodomo** [comment]
Let's wait in case maintainers would like to comment the discussion.

**Review by lsiepel — APPROVED** [review]

File: bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/discovery/AndroidTVMDNSDiscoveryParticipant.java (hunk 2 above)
Comment (lolodomo) [reply bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/discovery/AndroidTVMDNSDiscoveryParticipant.java]: Also @fwolter

Comment (wborn) [reply bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/discovery/AndroidTVMDNSDiscoveryParticipant.java]: If a `@Component` only implements one interface, bnd automatically adds that interface to the provided interfaces in the generated component XML files, e.g. `target/classes/OSGI-INF/org.openhab.binding.androiddebugbridge.internal.discovery.AndroidTVMDNSDiscoveryParticipant.xml `:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<scr:component xmlns:scr="http://www.osgi.org/xmlns/scr/v1.3.0" name="org.openhab.binding.androiddebugbridge.internal.discovery.AndroidTVMDNSDiscoveryParticipant" activate="activate" modified="modified" configuration-pid="discovery.androiddebugbridge">
  <service>
    <provide interface="org.openhab.core.config.discovery.mdns.MDNSDiscoveryParticipant"/>
  </service>
  <implementation class="org.openhab.binding.androiddebugbridge.internal.discovery.AndroidTVMDNSDiscoveryParticipant"/>
</scr:component>
```

So it is redundant if the list of `<provide interface=`  values is still correct if you remove the service from the `@Component`  annotation.

Sometimes the generated XML is wrong e.g. if your discovery service implements multiple interfaces or extends from the `AbstractDiscoveryService`... in that case it is mandatory to specify the list of service interfaces in the `@Component` annotation in order for openHAB Core to be able to use all implemented inferfaces.

Comment (lolodomo) [reply bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/discovery/AndroidTVMDNSDiscoveryParticipant.java]: Thank you @wborn for the clear explanation.

So it was not required to add it in the current case but this is also not an error to have it.

**Comment by lolodomo** [comment]
Let's merge that PR...

**Pull request merged**